packages/next/build/webpack-config.ts

```diff
@@ -623,17 +631,13 @@ export default async function getBaseWebpackConfig(
       return callback()
     }
 
-    let isNextExternal: boolean = false
-    if (isLocal) {
-      // we need to process next-server/lib/router/router so that
-      // the DefinePlugin can inject process.env values
-      isNextExternal = /next[/\\]dist[/\\]next-server[/\\](?!lib[/\\]router[/\\]router)/.test(
-        res
-      )
-
-      if (!isNextExternal) {
-        return callback()
-      }
+    // we need to process next-server/lib/router/router so that
+    // the DefinePlugin can inject process.env values
+    const isNextExternal = /next[/\\]dist[/\\]next-server[/\\](?!lib[/\\]router[/\\]router)/.test(
+      res
+    )
+    if (isLocal && !isNextExternal) {
+      return callback()
```

Comment:
Why was this code path altered? It looks functionally equivalent to before, but removed an important performance optimization.

We want to skip executing this regex when the if condition would've short circuited.

Reply:
The new behaviour is different because `isNextExternal` now has the correct value when `isLocal === false`, affecting the reference to `isNextExternal` on line 647 below. Previously, when `isLocal === false`, `isNextExternal` would always be `false` because line 630 would not be executed.

IIRC, e2e tests failed without this change, but I'll test this again.